2048.py

```diff
@@ -0,0 +1,275 @@
+        while row[index].value == 0 and row[index - 1].value > 0:
+            return True
+    return False
+
+
+def right_and_down_sort(array):
+    for index in range(len(array) - 1, 0, -1):
+        if array[index].value == 0 and array[index - 1].value > 0:
+            temp_index = array[index].value
+            temp_index_minus_one = array[index - 1].value
+            array[index - 1].value = temp_index
+            array[index].value = temp_index_minus_one
+
+
+def move_right(board):
+    for row in board:
+        while not_sorted_move_right_and_down(row):
+            right_and_down_sort(row)
+    return board
+
+
+def move_down(board):
+    for column in get_columns(board):
+        while not_sorted_move_right_and_down(column):
+            right_and_down_sort(column)
+    return board
+
+
+def right_or_down_sum(array):
+    for index in range(len(array) - 1, 0, -1):
+        if array[index].value == array[index - 1].value:
+            array[index].value += array[index - 1].value
+            array[index - 1].value = 0
+
+
+def sum_right(board):
+    for row in board:
+        right_or_down_sum(row)
+    return board
+
+
+def sum_down(board):
+    for column in get_columns(board):
+        right_or_down_sum(column)
+    return board
+
+
+def left_or_up_sum(array):
+    for index in range(0, len(array) - 1):
+        if array[index].value == array[index + 1].value:
+            array[index].value += array[index + 1].value
+            array[index + 1].value = 0
+
+
+def sum_left(board):
+    for row in board:
+        left_or_up_sum(row)
+    return board
+
+
+def sum_up(board):
+    for column in get_columns(board):
+        left_or_up_sum(column)
+    return board
+
+
+def get_fields_with_zero_value(board):
+    fields_with_zero_value = []
+    for row in board:
+        for field in row:
+            if field.value == 0:
+                fields_with_zero_value.append(field)
+    return fields_with_zero_value
+
+def place_next_element_on_board(board):
+    if len(get_fields_with_zero_value(board)) > 0:
+        next_field = random.choice(get_fields_with_zero_value(board))
+        next_field.value = 2
+    else:
+        print("No more space on the board. You lost!")
+
+
+def check_if_similar_neighboring(board):
+    for row in board:
+        for index in range(0, len(row)-1):
+            if row[index].value == row[index + 1].value:
+                return True
+    for column in get_columns(board):
+        for index in range(0, len(column)-1):
+            if column[index].value == column[index + 1].value:
+                return True
+    return False
+
+
+
+def game_over(board):
+    if len(get_fields_with_zero_value(board)) > 0:
+        return False
+
+
+def check_if_won(board):
+    winning_number = 16
+    for row in board:
+        for field in row:
+            if field.value == winning_number:
+                return True
+    return False
+
+
+def get_temporary_board_values(board):
```

Comment:
What is the difference between these two methods: `get_temporary_board_values` and `get_board_values`?

Reply:
One of these got deleted and correction in method previously using both of them was made.